File: python/7 kyu/395.py
# https://www.codewars.com/kata/58856a06760b85c4e6000055/train/python

# Passed

def bits_battle(numbers):
    evens_odds = [0, 0]
    for num in numbers:
        if num == 0:
            continue

        idx = num % 2
        count = bin(num).split('b')[-1].count(str(idx))
        evens_odds[idx] += count
    
    evens_count, odds_count = evens_odds
    if evens_count == odds_count:
        result = "tie"
    elif evens_count > odds_count:
        result = "evens win"
    else:
        result = "odds win"
        
    return result

output = bits_battle([3, 8, 22, 15, 78])
print(output)

File: nodejs/7 kyu/395.js
// https://www.codewars.com/kata/5a036ecb2b651d696f00007c/train/javascript

// Passed

function drawACross(n) {
  let result;
  if (n < 3) {
    result = "Not possible to draw cross for grids less than 3x3!";
  } else {
    let rows = [];
    let isCenteredCrossPossible = false;
    for (let rowIdx = 0; rowIdx < n; rowIdx++) {
      let row = " ".repeat(n).split("");
      let [startIdx, endIdx] = [rowIdx, n - rowIdx - 1];
      row[startIdx] = "x";
      row[endIdx] = "x";
      if (startIdx === endIdx) {
        isCenteredCrossPossible = true;
      }
      rows.push(row);
    }

    result = "Centered cross not possible!";
    if (isCenteredCrossPossible) {
      rows = rows.map((row) => row.join(""));
      result = rows.join("\n");
    }
  }

  return result;
}

const output = drawACross(7);
console.log(output);